src/main/java/ui/UI.java

```diff
@@ -555,13 +556,19 @@ public HWND getMainWindowHandle() {
         return mainWindowHandle;
     }
 
-    public void showNotificationPane(Node graphic, String text, Action action) {
+    public void showNotificationPane(Node graphic, String text, Runnable runnable, Runnable undoRunnable) {
```

Comment:
This also seems rather ad-hoc, because this method is for showing a notification in general, which may not have undo semantics. Again, I'm now sure how much of the architecture we should flesh out now.